panther_cli/main.py

```diff
@@ -170,6 +172,59 @@ def zip_policies(args: argparse.Namespace) -> Tuple[int, str]:
         'zip', args.policies)
 
 
+def upload_policies(args: argparse.Namespace) -> Tuple[int, str]:
+    """Tests, validates, packages, and then uploads all policies into a Panther deployment.
+
+    Returns 1 if the policy tests, validation, or packaging fails.
+
+    Args:
+        args: The populated Argparse namespace with parsed command-line arguments.
+
+    Returns:
+        A tuple of return code and the archive filename.
+    """
+    return_code, archive = zip_policies(args)
+    if return_code == 1:
+        return return_code, ''
+
+    client = boto3.client('lambda')
+
+    with open(archive, 'rb') as analysis_zip:
+        zip_bytes = analysis_zip.read()
+        payload = {
+            'resource':
+                '/upload',
+            'HTTPMethod':
+                'POST',
+            'Body':
+                json.dumps({
+                    'Data': base64.b64encode(zip_bytes).decode('utf-8'),
+                    'UserID': 'c273fd96-88d0-41c4-a74e-941e17832915',
+                }),
+        }
+
+        logging.info('Uploading pack to Panther')
+        response = client.invoke(FunctionName='panther-analysis-api',
+                                 InvocationType='RequestResponse',
+                                 LogType='None',
+                                 Payload=json.dumps(payload))
+
+        response_str = response['Payload'].read().decode('utf-8')
+        response_payload = json.loads(response_str)
+
+        if response_payload['statusCode'] != 200:
+            return 1, ''
+
+        body = json.loads(response_payload['body'])
+        logging.info('Upload success.')
+        logging.info(
+            '\n\t%d new policies\n\t%d modified policies\n\t%d new rules\n\t%d modified rules',
+            body['newPolicies'], body['modifiedPolicies'], body['newRules'],
+            body['modifiedRules'])
```

Comment:
Maybe we should just print the whole response?

```python
logging.info(json.dumps(body, indent=2, sort_keys=True))
```

Reply:
That would output the following:

```
[INFO]: Upload success.
[INFO]: {
  "modifiedPolicies": 13,
  "modifiedRules": 0,
  "newPolicies": 0,
  "newRules": 0,
  "totalPolicies": 13,
  "totalRules": 0
}
```

That looks reasonable, but I think I'll also specify that this is the response so users aren't confused by what this JSON blob is:

```
[INFO]: Upload success.
[INFO]: API Response:
{
  "modifiedPolicies": 13,
  "modifiedRules": 0,
  "newPolicies": 0,
  "newRules": 0,
  "totalPolicies": 13,
  "totalRules": 0
}
```